### PR TITLE
HAWQ-367. Introduce seg_max_connections guc to set segment max_connections setting

### PR DESCRIPTION
**Pull request by jiny2**

### CODE REVIEW

File: src/backend/utils/misc/guc.c [hunk 1]
```diff
@@ -4617,6 +4617,20 @@ static struct config_int ConfigureNamesInt[] =
...
+	 */
+	{
+		{"seg_max_connections", PGC_POSTMASTER, CONN_AUTH_SETTINGS,
+			gettext_noop("Sets the maximum number of concurrent connections in a segment."),
+			NULL
+		},
+		&SegMaxBackends,
+		1600, 80, MAX_MAX_BACKENDS, NULL, NULL
+	},
+
```
Comment (yaoj2) [inline src/backend/utils/misc/guc.c]: For default value, according to 1.x  the default is 3 times master value. So we use 200 as master value then 200*3 * 8 we use 4800 as default value.

Comment (jiny2) [inline src/backend/utils/misc/guc.c]: Ok, I fix it.

**Comment by huor** [comment]
LGTM. +1

**Comment by linwen** [comment]
Looks good

**Pull request closed without merging**